adapters/repos/db/roaringset/compactor.go

```diff
@@ -22,6 +22,74 @@ import (
 	"github.com/weaviate/weaviate/adapters/repos/db/lsmkv/segmentindex"
 )
 
+type compactorWriter interface {
+	segmentindex.SegmentWriter
+	Reset(io.Writer)
+}
+
+type memoryWriter struct {
+	buffer *[]byte
+	pos    *int
+	maxPos *int
+	writer io.WriteSeeker
+}
+
+// NewMemoryWriterWrapper creates a new memoryWriter with initialized pointers
+func newMemoryWriterWrapper(initialCapacity int64, writer io.WriteSeeker) *memoryWriter {
+	buf := make([]byte, initialCapacity)
+	pos := 0
+	maxPos := 0
+	return &memoryWriter{
+		buffer: &buf,
+		pos:    &pos,
+		maxPos: &maxPos,
+		writer: writer,
+	}
+}
+
+func (mw memoryWriter) Write(p []byte) (n int, err error) {
```

Comment:
question (non-blocking): related to previous comment about individual pointers. Why not make this a pointer receiver method and then no longer require individual pointers to all struct props?